src/librustdoc/clean/mod.rs

```diff
@@ -2132,7 +2133,7 @@ impl Clean<Item> for (&hir::ForeignItem<'_>, Option<Symbol>) {
                         decl,
                         generics,
                         header: hir::FnHeader {
-                            unsafety: hir::Unsafety::Unsafe,
+                            unsafety: intrinsic_operation_unsafety(item.ident.name),
```

Comment:
This should only apply for `extern "rust-intrinsic"`, not `extern "C"`.

Reply:
Good point, thanks for bringing it up!